proto/table_marshal.go

```diff
@@ -225,6 +225,9 @@ func (u *marshalInfo) marshal(b []byte, ptr pointer, deterministic bool) ([]byte
 	// If the message can marshal itself, let it do it, for compatibility.
 	// NOTE: This is not efficient.
 	if u.hasmarshaler {
+		if deterministic {
+			return nil, errors.New("proto: deterministic not supported by the Marshal method of " + u.typ.Name())
```

Comment:
s/Name/String/

`reflect.Type.Name` only prints the type name, while `reflect.Type.String` provides the package as well.

Reply:
Thanks for the suggestion.